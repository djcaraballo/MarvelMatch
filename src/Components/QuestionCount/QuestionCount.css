@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

.question-count {
  font-family: 'Orbitron', sans-serif;
  font-size: 1.5em;
  color: #fff;
}