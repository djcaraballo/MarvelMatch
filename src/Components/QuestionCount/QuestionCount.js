import React from 'react'
import PropTypes from 'prop-types'
import './QuestionCount.css'

const QuestionCount = ({ counter, total }) => {
  return (
    <div className="question-count">
      <h6>Question <span>{counter}</span> of <span>{total}</span></h6>
    </div>
  )
}

QuestionCount.propTypes = {
  counter: PropTypes.number,
  total: PropTypes.number
}

export default QuestionCount
