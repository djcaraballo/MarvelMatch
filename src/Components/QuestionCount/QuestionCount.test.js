import React from 'react'
import QuestionCount from './QuestionCount'
import { shallow } from 'enzyme'

describe('QuestionCount Component', () => {
  let wrapper
  let mockCounter = 0
  let mockTotal = 11

  beforeEach(() => {
    wrapper = shallow(
      <QuestionCount 
        counter={mockCounter}
        total={mockTotal}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})