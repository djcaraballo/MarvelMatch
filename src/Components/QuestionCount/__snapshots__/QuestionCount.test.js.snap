// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCount Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionCount
    counter={0}
    total={11}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h6>
        Question 
        <span>
          0
        </span>
         of 
        <span>
          11
        </span>
      </h6>,
      "className": "question-count",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Question ",
          <span>
            0
          </span>,
          " of ",
          <span>
            11
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Question ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
        " of ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 11,
          },
          "ref": null,
          "rendered": 11,
          "type": "span",
        },
      ],
      "type": "h6",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h6>
          Question 
          <span>
            0
          </span>
           of 
          <span>
            11
          </span>
        </h6>,
        "className": "question-count",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Question ",
            <span>
              0
            </span>,
            " of ",
            <span>
              11
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Question ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
          " of ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 11,
            },
            "ref": null,
            "rendered": 11,
            "type": "span",
          },
        ],
        "type": "h6",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
